pkg/istio/rbac/v2/provider_test.go

```diff
@@ -40,23 +40,76 @@ var (
+				"app": "productpage",
+			},
+		},
+	}
 )
 
 func TestConvertAthenzModelIntoIstioRbac(t *testing.T) {
-	athenzclientset := fakev1.NewSimpleClientset()
-	fakeAthenzInformer := adInformer.NewAthenzDomainInformer(athenzclientset, 0, cache.Indexers{})
-	signedDomain := getFakeDomain()
-	labels := onboardedService.GetLabels()
-	domainRBAC := athenz.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &fakeAthenzInformer)
+	tests := []struct {
+		name                string
+		inputAthenzDomain   zms.SignedDomain
+		inputService        *k8sv1.Service
+		expectedAuthzPolicy []model.Config
+	}{
+		{
+			name:                "should create expected authz policy spec",
+			inputAthenzDomain:   getFakeOnboardedDomain(),
+			inputService:        onboardedService,
+			expectedAuthzPolicy: getExpectedAuthzPolicy(),
+		},
+		{
+			name:                "should create empty authz policy spec for service which doesn't have roles / policies defined",
+			inputAthenzDomain:   getFakeNotOnboardedDomain(),
+			inputService:        undefinedAthenzRulesServiceWithAnnotationTrue,
+			expectedAuthzPolicy: getExpectedEmptyAuthzPolicy(),
+		},
+	}
 
-	p := NewProvider(false, true)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			athenzclientset := fakev1.NewSimpleClientset()
+			fakeAthenzInformer := adInformer.NewAthenzDomainInformer(athenzclientset, 0, cache.Indexers{})
+			labels := onboardedService.GetLabels()
+			domainRBAC := athenz.ConvertAthenzPoliciesIntoRbacModel(tt.inputAthenzDomain.Domain, &fakeAthenzInformer)
+			p := NewProvider(false, true)
+			convertedAuthzPolicy := p.ConvertAthenzModelIntoIstioRbac(domainRBAC, tt.inputService.Name, labels["app"])
+			assert.Equal(t, tt.expectedAuthzPolicy, convertedAuthzPolicy, "converted authz policy should be equal")
+		})
+	}
+}
 
-	convertedCR := p.ConvertAthenzModelIntoIstioRbac(domainRBAC, onboardedService.Name, labels["app"])
-	expectedCR := getExpectedCR()
-	assert.Equal(t, expectedCR, convertedCR, "converted authz policy should be equal")
+func getExpectedEmptyAuthzPolicy() []model.Config {
+	var out model.Config
+	schema := collections.IstioSecurityV1Beta1Authorizationpolicies
+	createTimestamp, _ := time.Parse("", "12/8/2015 12:00:00")
+	out.ConfigMeta = model.ConfigMeta{
+		Type:              schema.Resource().Kind(),
+		Group:             schema.Resource().Group(),
+		Version:           schema.Resource().Version(),
+		Namespace:         "test-namespace",
+		Name:              "onboarded-service",
+		CreationTimestamp: createTimestamp,
+	}
+	out.Spec = &v1beta1.AuthorizationPolicy{
+		Selector: &workloadv1beta1.WorkloadSelector{
+			MatchLabels: map[string]string{"svc": "productpage"},
+		},
+	}
+	return []model.Config{out}
 }
 
-func getExpectedCR() []model.Config {
+func getExpectedAuthzPolicy() []model.Config {
 	var out model.Config
 	schema := collections.IstioSecurityV1Beta1Authorizationpolicies
 	createTimestamp, _ := time.Parse("", "12/8/2015 12:00:00")
```

Comment:
leave as a placeholder, err should be checked from upstream branch, double check after pull from branch authzpolicy

Reply:
Added.